libraries/chain/db_update.cpp

```diff
@@ -388,6 +389,16 @@ void database::clear_expired_orders()
 
          price settlement_price = pays / receives;
 
+         // Calculate fill_price with a bigger volume to reduce impacts of rounding
+         if( settlement_fill_price.base.asset_id != pays.asset_id )
```

Comment:
Perhaps use current_asset instead of pays.asset_id to make it clear that this block is meant to initialize settlement_fill_price once per asset.

Reply:
Makes sense.